pkgs/applications/video/mpv/scripts/seek-to.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+let src = fetchFromGitHub {
+      owner = "occivink"; repo = "mpv-scripts";
+      rev = "af360f332897dda907644480f785336bc93facf1";
+      hash = "sha256-KdCrUkJpbxxqmyUHksVVc8KdMn8ivJeUA2eerFZfEE8=";
+    };
+    version = "unstable-2022-10-02";
+in
```

Comment:
```suggestion
let
  src = fetchFromGitHub {
    owner = "occivink";
    repo = "mpv-scripts";
    rev = "af360f332897dda907644480f785336bc93facf1";
    hash = "sha256-KdCrUkJpbxxqmyUHksVVc8KdMn8ivJeUA2eerFZfEE8=";
  };
  version = "unstable-2022-10-02";
in
```

Reply:
> Yes, I did that because I was considering packaging more scripts from the same repo, in which case factoring out src and version is useful.

You could create a occivink-mpv-generic.nix with a function that takes pname, etc. in this pr

Reply:
Done!  It seemed a bit silly to have it all in separate 2-line files, so I did something slightly different (and feedback is much appreciated ;3)

Reply:
please move the generic function into it's own file

do something like (don't use name as the argument name since that might be set to something...)

```
{ scriptName
, stdenvNoCC
}:

...mkDerivation rec {
}
```

and then just callPackage ./occivik-generic.nix { scriptName = "bla"; }  it in toplevel if it's really just the scriptName (otherwise create a seperate file if it's more (like extra patches) "{ callPackage }: callPackage ./occivik-generic.nix { scriptName = "...."; other = ... }```)

the unionOfDisjoint in the other file: doing it like this is weird and I'm unsure if it's side-effect free (so do it in toplevel)

or otherwise creating a scope like cinnamon does it https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/cinnamon/default.nix#L3 with all the packages is fine by me. (that's how that usually gets handled)

or feel free to look at existing code in nixpkgs and take inspiration

Reply:
I don't know what the nix/nixpkgs behaviour will be but if you still want to pass name, etc you could perhaps `{ stdenv... } @ args:` and then merge args into that. but not sure if it's going to pull all of nixpkgs in then.

Reply:
> and then just callPackage ./occivik-generic.nix { scriptName = "bla"; } it in toplevel if it's really just the scriptName 

Just the script name and description, yes.  The latter is what I didn't want to add to `all-packages.nix`, since almost nobody else did.

> the unionOfDisjoint in the other file: doing it like this is weird and I'm unsure if it's side-effect free (so do it in toplevel)

OK, replaced with `//`.

> or otherwise creating a scope like cinnamon does [...]with all the packages is fine by me. (that's how that usually gets handled)

I'm also fine making a scope if that's preferred.
Thanks a bunch for all the helpful feedback  <3